### ISSUE TITLE
When panel is locked, allow to configure individual widgets

### ISSUE THREAD

**Issue by goetzc** [lxde/lxqt#1087]
Like changing Date and Time settings, or the Keyboard Layout.

Also, when the panel is locked, changing the text "Lock This Panel" to "Unlock This Panel" can help with the usability.
![lxqt panel lock](https://cloud.githubusercontent.com/assets/2220440/16669298/523aaaba-445a-11e6-983b-3773b39fe1d8.png)

**Comment by palinek** [comment]
The purpose of "lock" is to prevent any (unintentional) changes of particular panel & its widgets => I don't agree with this request.

...but let's see what others will say.

**Comment by stefonarch** [comment]
Locked is "locked everything" for me too.

**Comment by pmattern** [comment]
I agree with @palinek and @stefonarch as well.

**Comment by goetzc** [comment]
I thought the propose of this was only to prevent the accidental widget's deletion or intentional position change or move.

It seems to me that allowing to edit the individual widgets settings is a more standard behavior than forbidding everything. 

Just imagine you want to change some settings like the keyboard layout or the clock's time zone, you unlock the panel, then right click the widget again to select the configure option, but you unintentionally hit the move or remove option. On the other hand, panel ia locked, you right click the widget and select configure. If you mishit it doesn't matter because the other options are disabled.

**Comment by goetzc** [comment]
> Locked is "locked everything" for me too.

Why should it lock everything? It seems to be more useful to only lock the panel's dimensions and it's elements positions.

**Comment by tsujan** [comment]
I also agree with @palinek, @stefonarch, and @pmattern. Locking means preventing any change to the panel.

@goetzc 
Maybe the best way for not losing the settings is creating a backup of `~/.config/lxqt`. No setting could guarantee that the user wouldn't make a mistake.

**Comment by stefonarch** [comment]
> Just imagine you want to change some settings like the keyboard layout or the clock's time zone, you unlock the panel, then right click the widget again to select the configure option, but you unintentionally hit the move or remove option. On the other hand, you unlock the panel, then you right click the widget and select configure. If you mishit it doesn't matter because the other options are disabled.

There is a point about that, but I think this happens very rarely, keyboard layout p.e.  is under keyboard settings and a user configures it usually only once.

**Comment by f2404** [comment]
Other DEs, e.g. Cinnamon that I'm currently running, do allow to configure the applets while the panel is locked.
And I think this makes sense as locking the panel means no changes can be made to the panel: i.e. no adding or removing applets, or moving existing ones. But, why should this mean no changes to the applets' configuration? Applets are separate entities; changing their configuration is not the same as modifying the panel.

**Comment by goetzc** [comment]
>keyboard layout p.e. is under keyboard settings and a user configures it usually only once.

True, this example is not the most common. Another like the clock's time zone, maybe the task manager or other widgets may be more likely to be changed.

> Maybe the best way for not losing the settings is creating a backup of ~/.config/lxqt. No setting could guarantee that the user wouldn't make a mistake.

Sure, I do have such backups. It is not for me, a non-tech savvy person, let's say a Lubuntu user (when Lubuntu upgrades to this new Qt version) might not do such backups.

> Locking means preventing any change to the panel.

Using the broadest meaning of lock may not necessarily be the best option IMO. What does a user obtain preventing _any_ change? It seems to me, preventing some changes and allowing others would be more useful. And the widgets settings are not changed on the fly (a new window appears), like moving a widget's position or completely removing it.

@all, what do you think, what would a user loose and gain when configuring a widget is always allowed?

**Comment by tsujan** [comment]
> Other DEs, e.g. Cinnamon that I'm currently running, do allow to configure the applets while the panel is locked.

Oh, that made me reconsider my previous statement. Also with KDE, it's possible to configure widgets when they're locked. I don't say we should follow KDE but, at least, the idea doesn't seem odd to me anymore ;)

**Comment by Vladimir-csp** [comment]
I my experience locking panel widgets means preventing their movement/addition/removal. But not blocking configuration of widgets themselves, that would be overkill.

**Comment by user20190106** [comment]
I stumbled on this issue when I tried to find out what ”Lock This Panel” is supposed to do. I had assumed that it would prevent e.g. accidentally dragging the panel to some other part of the screen as in Windows (at least Win7).

Having read the above and having done some little tests, I suggest that you **either change it so that it locks the panel configuration but allows to configure individual widgets** - - e.g. change date display format; change base directory of the Directory Menu; change how many entries are shown when searching from the main menu, etc -- as suggested by @goetzc  and @Vladimir-csp  above -- **or remove the menu option altogether**. 

**As it stands now,  the menu option is completely redundant:** if a panel is locked, I cannot even see how it is configured – I need to unlock first… ...which undermines the purpose the lock.

Besides, it does not affect QuickLaunch: one can remove an application from QuickLaunch or move it although the panel is locked. Which is confusing for a user: I can edit this part of panel but not that part.

**Comment by stefonarch** [comment]
I could change my mind now.
